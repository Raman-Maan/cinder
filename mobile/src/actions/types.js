export const MATCHES_ERROR = 'MATCHES_ERROR';
export const MATCHES_LOADING = 'MATCHES_LOADING';
export const MATCHES_FETCH_SUCCESS = 'MATCHES_FETCH_SUCCESS';
export const MATCHES_MATCHED = 'MATCHES_USERS_MATCHED';
export const MATCHES_REDIRECTED = 'MATCHEDS_USERS_MATCHED_REDIRECTED';

export const REC_ERROR = 'RECOMMENDATIONS_ERROR';
export const REC_LOADING = 'RECOMMENDATIONS_LOADING';
export const REC_FETCH_SUCCESS = 'RECOMMENDATIONS_FETCH_SUCCESS';
export const REC_SUBMIT_SUCCESS = 'RECOMMENDATIONS_SUBMIT_SUCCESS';
