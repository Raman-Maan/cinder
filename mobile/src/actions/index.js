// This is the action controller
// add all action files in here to be exported
// e.g. export * from './example'
// if unsure refer to web client for example
import * as types from './types';

export * from './matches';
export * from './recommendations';
export * from './ref';
export * from './userInfo';

export default types;
