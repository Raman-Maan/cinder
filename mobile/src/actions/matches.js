import types from '../actions';
import { serverURL } from '../../env';

export function matchesErrored(bool) {
  return {
    type: types.MATCHES_ERROR,
    error: bool,
  };
}

export function matchesLoading(bool) {
  return {
    type: types.MATCHES_LOADING,
    loading: bool,
  };
}

export function matchesFetchSuccess(data) {
  return {
    type: types.MATCHES_FETCH_SUCCESS,
    matches: data,
  };
}

export function matchesFetchData(userID) {
  return (dispatch) => {
    dispatch(matchesLoading(true));

    return fetch(`${serverURL}/api/users/${userID}/matches`)
      .then((res) => {
        if (!res.ok) {
          throw Error(res.statusText);
        }
        dispatch(matchesLoading(false));
        return res.json();
      })
      .then(data => dispatch(matchesFetchSuccess(data)))
      .catch(() => dispatch(matchesErrored(true)));
  };
}
