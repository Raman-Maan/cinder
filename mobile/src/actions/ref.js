import types from '../actions';
import { serverURL } from '../../env';

export function refErrored(bool) {
  return {
    type: types.REF_ERROR,
    error: bool,
  };
}

export function refLoading(bool) {
  return {
    type: types.REF_LOADING,
    loading: bool,
  };
}

export function refFetchSuccess(data) {
  return {
    type: types.REF_FETCHED,
    ref: data,
  };
}

export function refFetchData(type) {
  return (dispatch) => {
    dispatch(refLoading(true));

    return fetch(`${serverURL}/api/ref/${type}`)
      .then((res) => {
        if (!res.ok) {
          throw Error(res.statusText);
        }
        dispatch(refLoading(false));
        return res.json();
      })
      .then(data => dispatch(refFetchSuccess(data)))
      .catch(() => dispatch(refErrored(true)));
  };
}
