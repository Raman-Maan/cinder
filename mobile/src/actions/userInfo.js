import {
  USER_INFO_ERROR,
  USER_INFO_LOADING,
  USER_INFO_FETCH_SUCCESS,
} from './types';
import { serverURL } from '../../env';

function userInfoErrored(message) {
  return {
    type: USER_INFO_ERROR,
    payload: message,
  };
}

function userInfoLoading() {
  return {
    type: USER_INFO_LOADING,
  };
}

function userInfoFetchSuccess(data) {
  return {
    type: USER_INFO_FETCH_SUCCESS,
    payload: data,
  };
}

export function fetchUserInfo(userID) {
  return (dispatch) => {
    dispatch(userInfoLoading());
    fetch(`${serverURL}/api/users/${userID}`)
      .then(res => (res.ok ? res.json() : new Error(res.statusText)))
      .then((data) => {
        dispatch(userInfoFetchSuccess(data));
      })
      .catch(err => dispatch(userInfoErrored(err.message)));
  };
}
