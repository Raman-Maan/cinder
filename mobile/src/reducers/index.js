import { combineReducers } from 'redux';

const rootReducer = (state, action) => {}; //combineReducers({});

export default rootReducer;
