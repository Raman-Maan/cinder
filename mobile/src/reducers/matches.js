import * as types from '../actions/types';

export function matchesHasErrored(state = false, action) {
  switch (action.type) {
    case types.MATCHES_ERROR:
      return action.error;

    default:
      return state;
  }
}

export function matchesIsLoading(state = false, action) {
  switch (action.type) {
    case types.MATCHES_LOADING:
      return action.loading;

    default:
      return state;
  }
}

export function usersMatched(state = null, action) {
  switch (action.type) {
    case types.MATCHES_MATCHED:
      return action.matched;

    case types.MATCHES_REDIRECTED:
      return null;

    default: return state;
  }
}

export function matches(state = [], action) {
  switch (action.type) {
    case types.MATCHES_FETCH_SUCCESS:
      return action.matches;

    default:
      return state;
  }
}
