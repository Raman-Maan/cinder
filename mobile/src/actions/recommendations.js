import types from '../actions';
import { usersMatched } from './matches';
import { serverURL, loggedInUser } from '../../env';

export function recErrored(bool) {
  return {
    type: types.REC_ERROR,
    error: bool,
  };
}

export function recLoading(bool) {
  return {
    type: types.REC_LOADING,
    loading: bool,
  };
}

export function recFetchSuccess(recommendations) {
  return {
    type: types.REC_FETCH_SUCCESS,
    recommendations,
  };
}

export function recSubmitSuccess(result) {
  return {
    type: types.REC_SUBMIT_SUCCESS,
    result,
  };
}

export function fetchRecommendations() {
  return (dispatch) => {
    dispatch(recLoading(true));

    fetch(`${serverURL}/api/users/${loggedInUser}/recs`)
      .then((res) => {
        if (!res.ok) {
          throw Error(res.statusText);
        }
        dispatch(recLoading(false));
        return res.json();
      })
      .then(data => dispatch(recFetchSuccess(data)))
      .catch(() => dispatch(recErrored(true)));
  };
}

export function submitRecommendation(recID, action) {
  return (dispatch) => {
    fetch(`${serverURL}/api/users/${loggedInUser}/matches/${recID}/${action}`, {
      method: 'POST',
    })
      .then((res) => {
        if (!res.ok) {
          throw Error(res.statusText);
        }
        return res.json();
      })
      .then((data) => {
        dispatch(recSubmitSuccess(data));
        dispatch(usersMatched(data.matched === 'true'));
      })
      .catch(() => dispatch(recErrored(true)));
  };
}
