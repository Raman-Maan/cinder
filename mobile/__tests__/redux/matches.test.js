import mockStore from '../_config/mockStore';

import { matchesFetchData, matchesLoading, matchesErrored, matchesFetchSuccess } from '../../src/actions/matches';
import { matches, matchesHasErrored, matchesIsLoading } from '../../src/reducers/matches';

const initialState = {
  loading: false,
  error: false,
  matches: [],
};

describe('Reducers::Matches', () => {
  const store = mockStore(initialState);

  it('should respond to loading action', () => {
    expect(matchesIsLoading(initialState, matchesLoading(true))).toEqual(true);
    expect(matchesIsLoading(initialState, matchesLoading(false))).toEqual(false);
  });

  it('should respond to error action', () => {
    expect(matchesHasErrored(initialState, matchesErrored(true))).toEqual(true);
    expect(matchesHasErrored(initialState, matchesErrored(false))).toEqual(false);
  });

  it('should respond to matches actions', () => {
    const data = [1, 2, 3];
    expect(matches(initialState, matchesFetchSuccess(null))).toEqual(null);
    expect(matches(initialState, matchesFetchSuccess(data))).toEqual(data);
  });
});

describe('Actions::Matches', () => {
  const store = mockStore(initialState);

  beforeEach(() => {
    store.clearActions();
  });

  it('should handle fetching data - send correct actions', async () => {
    const response = '{"data": []}';
    fetch.mockResponseSuccess(response);
    await store.dispatch(matchesFetchData(10));

    expect(store.getActions()).toMatchSnapshot();
  });

  it('should handle fetching data - getting an error', async () => {
    fetch.mockResponseFailure(new Error('I am a bad bad error'));
    await store.dispatch(matchesFetchData(10));

    expect(store.getActions()).toMatchSnapshot();
  });
});

