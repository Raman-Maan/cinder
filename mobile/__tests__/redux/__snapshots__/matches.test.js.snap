// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions::Matches should handle fetching data - getting an error 1`] = `
Array [
  Object {
    "loading": true,
    "type": "MATCHES_LOADING",
  },
  Object {
    "error": true,
    "type": "MATCHES_ERROR",
  },
]
`;

exports[`Actions::Matches should handle fetching data - send correct actions 1`] = `
Array [
  Object {
    "loading": true,
    "type": "MATCHES_LOADING",
  },
  Object {
    "loading": false,
    "type": "MATCHES_LOADING",
  },
  Object {
    "matches": Object {
      "data": Array [],
    },
    "type": "MATCHES_FETCH_SUCCESS",
  },
]
`;

exports[`Actions::Matches should handle the result of user submissions - did match 1`] = `
Array [
  Object {
    "loading": true,
    "type": "MATCHES_LOADING",
  },
  Object {
    "matched": true,
    "type": "MATCHES_USERS_MATCHED",
  },
  Object {
    "loading": false,
    "type": "MATCHES_LOADING",
  },
]
`;

exports[`Actions::Matches should handle the result of user submissions - did not match 1`] = `
Array [
  Object {
    "matched": false,
    "type": "MATCHES_USERS_MATCHED",
  },
]
`;
